src/gui/widgets/template_list_widget.cpp

```diff
@@ -121,21 +121,13 @@ QVariant makeCheckBoxDecorator(QStyle* style, const QSize& size)
 	return pixmap;
 }
 
-/**
- * Returns a new QToolButton with a unified appearance.
- */
+/// Local wrapper for newToolButton adding the What's This bit.
 QToolButton* newToolButton(const QIcon& icon, const QString& text)
 {
-	auto* button = new QToolButton();
-	button->setToolButtonStyle(Qt::ToolButtonIconOnly);
-	button->setToolTip(text);
-	button->setIcon(icon);
-	button->setText(text);
-	button->setWhatsThis(OpenOrienteering::Util::makeWhatThis("templates.html#setup"));
-	return button;
+	return OpenOrienteering::Util::Tools::newToolButton(icon, text, "templates.html#setup");
 }
 
-}
+}  // anonymous namespace
```

Comment:
How about .. Well, I see, the namespace was already there :smile:

Reply:
One thing I hate about these reviews is that unheard-before rules appear in the course of the discussion. I acknowledge that I'm no more than an average coder and I know there are always things to improve. However, no other maintainer ever dug that deep into my code, with the exception of Junio H., who was right on every point like you often are. Adhering to the Coding Conventions and the tool suggestions does not give a guarantee that we will not spend time on technicalities over and over again. My wish is to learn this project's coding style without the need for the eternal patch rewrites.

Ad the placement of anonymous namespace, you spotted yourself that it's inconsistent throughout the existing code. Most of the time anonymous namespaces are enclosed in `OpenOrienteering` and a few times it's outside. As far as unqualified symbol lookup is concerned, the exact placement plays almost no role. The symbol scope is restricted to the translation unit anyway, so we are talking about the availability of the enclosed symbols outside the `OpenOrienteering` curly braces and potential shadowing of symbols defined outside of the OO namespace. The **consistency point is relevant** in my view but otherwise, the discussion feels very academic to me as I see only marginal benefits from either case. If I missed some crucial moment, I'll be happy to learn where I made the mistake.

Reply:
I intentionally did mark the PR as "comments", not "changes requested".
I'm sorry but I lack experience of receiving reviews, apart from contribution to some open source projects. It is a pity that I seem to have failed again to carry what is important, what is a sidenote, or what I learned myself from a review. Unless there is another tool then these comments, I must leave more thoughts unsaid.

With the namespace, the whole point is whether one writes / a future developer reads:

~~~
  return OpenOrienteering::Util::Tools::newToolButton(icon, text, "color_dock_widget.html");
~~~

vs.

~~~
  return Util::Tools::newToolButton(icon, text, "color_dock_widget.html");
~~~

Reply:
Yes, the path to `OpenOrienteering` symbols is another valid point for moving the namespace. Thanks!

Regarding the content of review feedback, I'd be happy with comments prefixed "side note", "optional change", "for your consideration" or similar. Chances are, I'll adopt most of your proposals because they are useful. It is the lack of clarity *when* the code change is suitable for merge that often drains my enthusiasm.